tests/lib/prepare.sh

```diff
@@ -456,8 +456,11 @@ uc20_build_initramfs_kernel_snap() {
         # this works on 20.04 but not on 18.04
         unmkinitramfs initrd unpacked-initrd
 
-        # use distro skeleton
-        cp -ar /usr/lib/ubuntu-core-initramfs skeleton
+        # use only the initrd we got from the kernel snap to inject our changes
+        # we don't use the distro package because the distro package may be 
+        # different systemd version, etc. in the initrd from the one in the 
+        # kernel and we don't want to test that, just test our snap-bootstrap
+        cp -ar unpacked-initrd skeleton
```

Comment:
Just a note, I recall that when using the skeleton from the unpacked initrd, the system would panic during boot. I suspected there was some symlinks (or hardlinks present in the distro package) were lost when unpacking, and the repacked image was not identical to one you'd obtain with the distro skeleton.